core/node/node_framework/src/resource/resource_collection.rs

```diff
@@ -104,3 +104,64 @@ impl<T: Resource + Clone> ResourceCollection<T> {
         (*handle).clone()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use tokio::time::timeout;
+
+    use super::*;
+
+    #[derive(Debug, Clone, PartialEq)]
+    struct TestResource(Arc<u8>);
+
+    impl Resource for TestResource {
+        fn resource_id() -> ResourceId {
+            ResourceId::new("test_resource")
+        }
+    }
+
+    #[tokio::test]
+    async fn test_push() {
```

Comment:
Probably it makes sense to ensure that we can push several different items to the collection.